### PR TITLE
treefmt: replace black with ruff

### PR DESCRIPTION
**Pull request by Mic92**
One tool less to download.
Ruff's formatting tries to stay close to 100% compatible with black while being faster.
Having one tool for everything also should in theory cause less compatibility issues.                                                

## Checklist for things done

- [ ] Summary of the proposed changes in the PR description
- [ ] More detailed description in the commit message(s)
- [ ] Commits are squashed into relevant entities - avoid a lot of minimal dev time commits in the PR
- [ ] [Contribution guidelines](https://github.com/tiiuae/ghaf/blob/main/CONTRIBUTING.md) followed
- [ ] Ghaf documentation updated with the commit - https://tiiuae.github.io/ghaf/
- [ ] PR linked to architecture documentation and requirement(s) (ticket id)
- [ ] Test procedure described (or includes tests). Select one or more:
  - [ ] Tested on Lenovo X1 `x86_64`
  - [ ] Tested on Jetson Orin NX or AGX `aarch64`
  - [ ] Tested on Polarfire `riscv64`
- [ ] Author has run `nix flake check --accept-flake-config` and it passes
- [ ] All automatic Github Action checks pass - see [actions](https://github.com/tiiuae/ghaf/actions)
- [ ] Author has added reviewers and removed PR draft status

### CODE REVIEW

File: nix/treefmt.nix [hunk 1]
```diff
@@ -24,13 +24,13 @@
         # Python
         # It was found out that the best outcome comes from running mulitple
         # formatters.
-        black.enable = true; # The Classic Python formatter
         isort.enable = true; # Python import sorter
         # Ruff, a Python formatter written in Rust (30x faster than Black).
         # Also provides additional linting.
         # Do not enable ruff.format = true; because then it won't complaing
```
Comment (Mic92) [inline nix/treefmt.nix:30]: Ah I didn't read this. I don't see any linting errors complaints though? So maybe this has been fixed?

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+[tool.ruff]
+line-length = 88
+target-version = "py311"
+lint.select = [ "E", "F", "I", "U", "N", "RUF", "A" ]
```
Comment (Mic92) [inline pyproject.toml:4]: I stands for isort

File: modules/jetpack/nvidia-jetson-orin/mk-esp-contents.py [hunk 3]
```diff
@@ -85,7 +86,7 @@ def copy_file(src: str, dst: str) -> None:
 def make_efi_name(store_file_path: str, root: str = "/") -> str:
     suffix = os.path.basename(store_file_path)
     store_dir = os.path.basename(os.path.dirname(store_file_path))
-    return os.path.join(root, "EFI/nixos/%s-%s.efi" % (store_dir, suffix))
+    return os.path.join(root, f"EFI/nixos/{store_dir}-{suffix}.efi")
```
Comment (Mic92) [inline modules/jetpack/nvidia-jetson-orin/mk-esp-contents.py:89]: This is the only line done manually. The rest was handled by ruff.

File: nix/treefmt.nix [hunk 4]
```diff
@@ -22,15 +22,9 @@
         statix.enable = true; # prevents use of nix anti-patterns https://github.com/nerdypepper/statix
 
         # Python
-        # It was found out that the best outcome comes from running mulitple
-        # formatters.
-        black.enable = true; # The Classic Python formatter
-        isort.enable = true; # Python import sorter
-        # Ruff, a Python formatter written in Rust (30x faster than Black).
-        # Also provides additional linting.
-        # Do not enable ruff.format = true; because then it won't complaing
```
Comment (Mic92) [inline nix/treefmt.nix:31]: I believe this is a past issue. I use this combination in many problems i.e. a large project called buildbot and have seen no issues.

**Review by brianmcgillion — APPROVED** [review]

**Pull request merged**